Prompt default office selection after user signs in for the first time

ThuTrinh opened the pull request:
# Description
When a user signs in for the first time, their default office is set for them. This can cause confusion later down the track if they don't notice it.

Now, a welcome notification will appear with instructions to change the default office if they wish. The user can dismiss this notification and it won't appear again.

## References
[Github Issue](https://github.com/zendesk/volunteer_portal/issues/67)

## Implementation notes (if needed)
This PR also introduces:

### [Apollo Hooks](https://www.apollographql.com/docs/react/api/react-hooks/) 
Apollo Hooks Provider can live alongside the old ApolloProvider. It is a much more terse interface. Hopefully this can also help us with moving away from redux.

```javascript
import { useMutation } from '@apollo/react-hooks'
import ConfirmProfileSettingsMutation from './mutations/confirmProfileSettings.gql'

const ConfirmProfileAlert = (_props) => {
  const [confirmProfileSettings, _] = useMutation(ConfirmProfileSettingsMutation)

  return(<button onClick={() => confirmProfileSettings()}>Click Me</button>)
}
```

### UserContext
A custom context that stores the current logged in user, provided at login. Subsequent components can access this context through the `useContext` hook. No longer needing to relying on graphql queries or prop passing. This lets us decouple out components as well as minimise unnecessary queries.

```javascript
import React, { useContext } from 'react'
import { UserContext } from '../../context'

const UserInfo = () => {
  const currentUser = useContext(UserContext)

  return (<div>{currentUser.name}</div>)
}
```


## Screenshots
![profile-setting-prompt](https://user-images.githubusercontent.com/3032977/72402728-1eda2b00-37a4-11ea-9056-cfe489d24ade.png)

## CCs
@zendesk/volunteer

## Risks (if any)
* [low] - all additive

ThuTrinh committed:
:construction: Record last_sign_in_at and current_sign_in_at timestamps

ThuTrinh committed:
:construction: Expose last_sign_in_at property on user

Use this to determin first logged in state

ThuTrinh committed:
:construction: Add user_preferences table

Use the user_preference table to store whether they have confirmed that their profile settings are correct. This is used to decide if we should show a prompt in the UI.

ThuTrinh committed:
:construction: Expose user_preference through graphql

ThuTrinh committed:
:heavy_plus_sign: Add apollo hooks and garden grids

Apollo Hooks can live alongside the old ApolloProvider. We want to start replacing the existting apollo queries and mutations with the new hooks implementation. It is a much more terse interface. Hopefully this can also help us with moving away from redux.

ThuTrinh committed:
:construction: Add user preference mutation to set confirmedProfileSettings

ThuTrinh committed:
:sparkles: Add UserContext and Notifications component to App

- UserContext stores the current logged in user, provided at login. Subsequent components can simply pull this out of context rather than relying on graphql queries or prop passing. This lets us decouple out components as well as minimise unnecessary queries.
- Notifications component is responsible for displaying notifications below the header.

ThuTrinh committed:
:white_check_mark: Add UserPreference model unit test

ThuTrinh committed:
:white_check_mark: Add user_preference_resolver_test

ThuTrinh committed:
:art: Rubocop

samzx commented on app/javascript/components/Notifications/index.js:
```diff
@@ -0,0 +1,37 @@
+import '@zendeskgarden/react-grid/dist/styles.css'
+import '@zendeskgarden/react-notifications/dist/styles.css'
+
+const isFirstSignIn = R.propSatisfies(R.isNil, 'lastSignInAt')
```
is there something we can do so user's that've already sign in before and have already changed their office won't get this prompt? It _may_ be noisy for those who've already set their office.

ThuTrinh replied:
hmm, I can backfill the user preference value for all existing users

samzx reviewed (commented):
The Apollo hooks and context are really nice 😍

ThuTrinh commented on app/javascript/components/Notifications/ConfirmProfileAlert.js:
```diff
@@ -0,0 +1,39 @@
+
+  if (show) {
+    return (
+      <Notification type="info">
```
I was tossing between modal and notification. I don't really mind either way. The benefit of the modal is that it is more in your face :)

ThuTrinh commented on app/graphql/types/user_preference_graph_type.rb:
```diff
@@ -0,0 +1,12 @@
+module Types
+  class UserPreferenceGraphType < Types::BaseObject
+    graphql_name 'UserPreference'
+    description 'An office'
```
woops

ThuTrinh committed:
:ok_hand: Correct UserPreferenceGraphType description

ThuTrinh committed:
:rewind: Add migration to backfill user_preferences for existing users

Set the confirmed_profile_settings preference to true for all existing users. They don't need to see the notification for this.

ThuTrinh committed:
:ok_hand: Switch from welcome notification to modal

In this case the modal makes more sense as we want the user to actively confirm they are happy with the assigned default office

ThuTrinh committed:
:white_check_mark: Update jest snapshots
